public/src/components/channelManagement/hooks/useValidation.ts

```diff
@@ -0,0 +1,30 @@
+import { useState, useEffect } from "react";
+import { ValidationStatus } from "../helpers/validation";
+
+const allValid = (validationStatus: ValidationStatus) => {
+  return Object.values(validationStatus).every((status) => status);
+};
+
+const useValidation = (onValidationChanged: (isValid: boolean) => void) => {
+  const [validationStatus, setValidationStatus] = useState<ValidationStatus>(
+    {}
+  );
+
+  const [wasValid, setWasValid] = useState(true);
+
+  useEffect(() => {
+    const isValid = allValid(validationStatus);
+    if (isValid !== wasValid) {
```

Comment:
is this extra `wasValid` state for performance reasons? Does it change the behaviour of the app?

Reply:
discussed IRL, it was triggering an infinite loop without this